src/compiler/emitter.ts

```diff
@@ -4046,7 +4006,7 @@ module ts {
                 var preambleEmitted = writer.getTextPos() !== outPos;
 
                 if (!preambleEmitted && nodeEndIsOnSameLineAsNodeStart(body, body)) {
-                    for (var i = 0, n = body.statements.length; i < n; i++) {
+                    for (var i = startIndex, n = body.statements.length; i < n; i++) {

```

Comment:
i guess this isn't really necessary.  If start index were not zero, then preamble emitted would be true.